src/blockchain/transactions.ts

```diff
@@ -137,37 +147,37 @@ export function send(
   }
 
   const result: Observable<TxState> = bindNodeCallback(method)(...args).pipe(
-    mergeMap((txHash: string) => {
-      return onEveryBlock$.pipe(
-        mergeMap(() => bindNodeCallback(web3.eth.getTransactionReceipt)(txHash)),
-        filter(receipt => !!receipt),
-        // to prevent degenerated infura response...
-        // tap((receipt: any) =>
-        //   console.log('receipt', receipt, receipt.blockNumber)
-        // ),
-        filter((receipt: any) =>
-          receipt.blockNumber !== undefined && receipt.blockNumber !== null
+    mergeMap(txHash => bindNodeCallback(web3.eth.getTransaction)(txHash)),
+    mergeMap((transaction: { hash: string, nonce: number, input: string }) => {
+      return combineLatest(externalNonce2tx$, onEveryBlock$).pipe(
+        map(([externalNonce2tx]) =>
+          externalNonce2tx[transaction.nonce] ?
+            [externalNonce2tx[transaction.nonce].hash, transaction.input === externalNonce2tx[transaction.nonce].callData ? TxRebroadcastStatus.speedup : TxRebroadcastStatus.cancel] :
+            [transaction.hash, undefined]
+        ),
+        mergeMap(([txHash, rebroadcast]) => bindNodeCallback(web3.eth.getTransactionReceipt)(txHash).pipe(map(receipt => [receipt, rebroadcast]))),
+        filter(([receipt]: [any, TxRebroadcastStatus]) =>
+          receipt && receipt.blockNumber !== undefined && receipt.blockNumber !== null
         ),
-        // tap(receipt => console.log('filtered receipt', receipt)),
-        take(1),
-        mergeMap(receipt => successOrFailure(txHash, receipt)),
+        first(),
+        mergeMap(([receipt, rebroadcast]) => successOrFailure(receipt.transactionHash, receipt, rebroadcast)),
         catchError(error => {
           return of({
             ...common,
-            txHash,
             error,
+            txHash: transaction.hash,
             end: new Date(),
             lastChange: new Date(),
             status: TxStatus.Error,
           } as TxState);
         }),
         startWith({
           ...common,
-          txHash,
+          txHash: transaction.hash,
           broadcastedAt: new Date(),
           status: TxStatus.WaitingForConfirmation,
         } as TxState),
-      );
+      ) as any as Observable<TxState>;
```

Comment:
Why not as `Observable<TxState>` directly? That double casting doesn't make sense to me

Reply:
Without cast it fails in my IDE.
With only `as Observable<TxState>` it complains about unnecessary cast in `yarn test`.
So this form is the only working everywhere.

Reply:
What IDE do you use?

Reply:
Atom, but webpack has the same problem